### ISSUE TITLE
Plugin options to support more use cases

### ISSUE THREAD

**Issue by khang-nd** [boyum/markdown-it-image-size#756]
Thanks for the great project and support.

I would like to suggest adding a few options for the plugin:

* `remote`: option to make external image requests optional (`false` by default). Reason is that SSG frameworks like 11ty or Vitepress have to process all images on the entire site everytime during development due to hot reload and having to fetch remote images everytime can significantly slow down the development speed if the internet connection is not in an ideal state.
* `publicDir`: option to allow local absolute path processing. This will replace the `"."` below with the specified public directory path.
https://github.com/boyum/markdown-it-image-size/blob/6ee9922e7034229c97419738d5bd6244a8d6f667/packages/markdown-it-image-size/src/index.ts#L39

Example:

```js
const MarkdownIt = require("markdown-it");
const { markdownItImageSize } = require("markdown-it-image-size");
const { cwd } = require("process");

const mdRenderer = MarkdownIt();
mdRenderer.use(markdownItImageSize, {
	publicDir: cwd() + "/public",
	remote: false
});
```

**Comment by khang-nd** [comment]
The idea of the `cache` option sounds brilliant. If we could realize it and make it a default behavior, I believe the `remote` option would not be necessary.

Regarding the `publicDir` suggestion, it actually came out from the fact that I was not using 11ty but Vitepress for my website, I wasn't able to use the package but had to pull the code and make appropriate modifications for it to work for my use case. Thought it'd be good to contribute back to the author, since there are actually some demands. (ref https://github.com/vuejs/vitepress/discussions/1639, https://github.com/vuejs/vitepress/discussions/3738)

**Comment by boyum** [comment]
#758 should add a file system cache, but I haven't really tried it yet (apart from unit tests) and I'm haven't done anything similar before. Please have a look at that, too (if you want, have time and care for it, no pressure).

**Comment by khang-nd** [comment]
Thank you for making the changes, I'll check them out later today or tomorrow and get back to you.

**Comment by boyum** [comment]
File system cache released in v14.4.0. I improved the unit tests and trust the code a bit more.
